### PR TITLE
Scope log path fix

### LINKED ISSUE #150: Logs inside app folder when using Buildkite
The agent is writing the logs inside the app folder when tests are executed inside Buildkite

### PR DESCRIPTION
**Pull request by tonyredondo**
Closes #150

### CODE REVIEW

**Comment by tonyredondo** [comment]
Ready for review @fermayo

File: agent/agent.go [hunk 1]
```diff
@@ -401,9 +401,13 @@ func getLogPath() (string, error) {
...
 		isOk := true
 		// If folder doesn't exist we try to create it
-		if _, err := os.Stat(logFolder); os.IsNotExist(err) {
-			mkErr := os.Mkdir(logFolder, os.ModeDir)
-			if mkErr != nil {
+		if _, err := os.Stat(logFolder); err != nil {
+			if os.IsNotExist(err) {
+				mkErr := os.Mkdir(logFolder, os.ModeDir)
+				if mkErr != nil {
+					isOk = false
```
Comment (fermayo) [inline agent/agent.go]: Why use `isOk` instead of just returning?

File: agent/agent.go [hunk 2]
```diff
@@ -401,9 +401,13 @@ func getLogPath() (string, error) {
 	if logFolder != "" {
 		isOk := true
 		// If folder doesn't exist we try to create it
-		if _, err := os.Stat(logFolder); os.IsNotExist(err) {
-			mkErr := os.Mkdir(logFolder, os.ModeDir)
-			if mkErr != nil {
+		if _, err := os.Stat(logFolder); err != nil {
+			if os.IsNotExist(err) {
+				mkErr := os.Mkdir(logFolder, os.ModeDir)
+				if mkErr != nil {
```
Comment (fermayo) [inline agent/agent.go]: This can be done in one line

File: agent/agent.go [hunk 3]
```diff
@@ -413,11 +417,18 @@ func getLogPath() (string, error) {
...
 	// If the log folder can't be used we return a temporal path, so we don't miss the agent logs
-	if dir, err := ioutil.TempDir("", "scope"); err != nil {
-		return dir, nil
-	} else {
-		return "", err
+	dir := path.Join(os.TempDir(), "scope")
+	if _, err := os.Stat(dir); err != nil {
+		if os.IsNotExist(err) {
+			mkErr := os.Mkdir(dir, os.ModeDir)
+			if mkErr != nil {
```
Comment (fermayo) [inline agent/agent.go]: This can be done in one line

File: go.mod [hunk 4]
```diff
@@ -14,10 +14,13 @@ require (
 	github.com/stretchr/testify v1.4.0
 	github.com/undefinedlabs/go-mpatch v0.0.0-20200122175732-0044123dbb98
 	github.com/vmihailenco/msgpack v4.0.4+incompatible
+	golang.org/x/lint v0.0.0-20190313153728-d0100b6bd8b3 // indirect
```
Comment (fermayo) [inline go.mod]: Avoid committing changes to `go.mod` like this, which are non-related to the PR

**Pull request merged**